### PR TITLE
First implementation of dashboard

### LINKED ISSUE #16: First Implementation of Dashboard
Inside the components folder, create the following components:

Refer to: https://www.figma.com/file/QZ0QibglxiI8UrFtSzclw0/OKB?node-id=281%3A149&t=sAVv8Fls2gNy3q5h-1
Image

a. `Dashboard.tsx`: This is the main dashboard component that will include all other components.

b. `AppointmentCard.tsx`: A component to display individual appointment cards. This component would likely take in the appointment data as props. You can define an interface appropriately.

c. `PsychiatristCard.tsx`: A component to display psychiatrist cards. This component would likely take in the psychiatrist's data and size of the component as props. You can define an interface appropriately. We take in size so that we don't have to define separate components for differently sized psychiatrist cards.
d. `ArticleCard.tsx`: A component to display article cards. For now, this will take in article metadata as props. We might have to use the Sanity CDN to load a preview of articles that could be linked to the old website. This is TBD. 

Note that these are all rough implementations of Mid-fis. Please do your best, be reasonable, and ask clarifying questions!

- [x] MOVE THIS CARD ON BOARD
- [x] Complete `AppointmentCard.tsx`
- [x] and `PsychiatristCard.tsx`
- [x] and `ArticleCard.tsx`
- [x] and `Dashboard.tsx`
- [x] MOVE
- [x] PR and Link PR

### PR DESCRIPTION
**Pull request by jessicajcho**
## Overview
resolves https://github.com/cornellh4i/okb-hope/issues/16
Created Appointment cards, Psychiatrist cards, and Article cards for dashboard view.

## Changes Made
- Created components for appointment, psychiatrist profile, and articles in AppointmentCard.tsx, PsychiatristCard.tsx, ArticleCard.tsx
- Added dummy components to Dashboard.tsx
- Currently no button functionality 

## Test Coverage

Manual testing. 

## Screenshots (delete if not applicable)

<details>

  <summary>Appointment and Psychiatrist Cards </summary>

  
image

</details>

<details>

  <summary>Article Cards </summary>

  
image

</details>

### CODE REVIEW

File: frontend/next.config.js [hunk 1]
```diff
@@ -3,6 +3,16 @@ const { default: next } = require('next')
...
+
+  webpack (config) {
+    config.module.rules.push({
+      test: /\.svg$/i,
+      issuer: /\.[jt]sx?$/,
+      use: ['@svgr/webpack']
+    })
+
+    return config
+  }
```
Comment (sriramanved) [inline frontend/next.config.js]: Why did you install webpack here and what do you need it for?

File: frontend/package.json [hunk 2]
```diff
@@ -9,23 +9,33 @@
...
     "eslint-config-next": "13.2.4",
     "next": "13.2.4",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "typescript": "4.9.5"
+    "react-router-dom": "^6.9.0",
+    "react-smart-flexbox": "^0.0.7",
+    "styled-components": "^5.3.9",
+    "typescript": "4.9.5",
+    "webpack": "^5.76.3"
```
Comment (sriramanved) [inline frontend/package.json]: Why are you guys using react-router in a NextJS project? You should not need this for routing. You should be consulting the documentation I provided to you guys.

File: frontend/src/components/AppointmentCard.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          <div className="grid grid-cols-5 grid-rows-2 gap-4 items-center pb-1/12">
+            <div className="col-span-1 shrink"><CalendarIcon></CalendarIcon></div>
+            <div className="col-span-4"><p>{day}, {month} {start.getDay()}</p></div>
+            <div className="col-span-1 shrink"><ClockIcon></ClockIcon></div>
+            <div className="col-span-4"><p>{start.getHours()}:{start.getMinutes()} - {end.getHours()}:{end.getMinutes()}</p></div>
+          </div>
+
+          <div className="card-actions justify-center">
+            <button className="btn w-7/12" >More Info</button>
+
```
Comment (sriramanved) [inline frontend/src/components/AppointmentCard.tsx]: should leave some comments here so other people can read your code.

File: frontend/src/components/ApptDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+//         <div>
+//           <p> Loading... </p>
+//         </div>
+//       )
+//     }
+//     if (this.state.error || !this.state.heroes) {
+//       return (
+//         <div>
+//           <p> An error occured </p>
+//         </div>
```
Comment (sriramanved) [inline frontend/src/components/ApptDetails.tsx]: why is there so much commented code left in a PR? Please remove this if its not being used.

File: frontend/src/components/ArticleCard.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React, { ReactNode, useState } from 'react'
+import ArticleIcon from '../assets/article.svg'
+
+
+const ArticleCard = ({ title, body }: { title: string, body: string }) => {
+  return (
+    <div className="card w-11/12 bg-base-100 shadow-xl">
+      <div className="card-body justify-center">
+        <ArticleIcon></ArticleIcon>
```
Comment (sriramanved) [inline frontend/src/components/ArticleCard.tsx]: You should only be returning a <ArticleIcon /> here. See PR made by @DylanTom and @hahnandrew if you need help.

File: frontend/src/components/Dashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        <h1 className="card-title pt-1/15">Upcoming Appointments</h1>
+          <div className="grid grid-cols-4 gap-4 items-center pb-1/12 shrink">
+            <AppointmentCard p_name={names[0]} start={new Date(starts[0])} end={new Date(ends[0])} description={dummy_descrip}>
+            </AppointmentCard>
+            <AppointmentCard p_name={names[1]} start={new Date(starts[1])} end={new Date(ends[1])} description={dummy_descrip}>
+            </AppointmentCard>
+            <AppointmentCard p_name={names[2]} start={new Date(starts[2])} end={new Date(ends[2])} description={dummy_descrip}>
+            </AppointmentCard>
+            <AppointmentCard p_name={names[3]} start={new Date(starts[3])} end={new Date(ends[3])} description={dummy_descrip}>
+            </AppointmentCard>
```
Comment (sriramanved) [inline frontend/src/components/Dashboard.tsx]: You should not be listing out every single one of the Appointment's here. For instance, what if there are less than 4 appointments to be displayed. You should be using .map() instead to return each of the Appointment cards.

File: frontend/src/components/Dashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import App from 'next/app';
+import React from 'react';
+import AppointmentCard from './AppointmentCard';
+import PyschiatristCard from './PsychiatristCard';
+import ArticleCard from './ArticleCard';
+
+const style = require("@/styles/Home.module.css");
```
Comment (sriramanved) [inline frontend/src/components/Dashboard.tsx]: why do you need this here?

File: frontend/src/components/Dashboard.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    </div>
+
+    <div className="card w-full bg-base-100 shadow-xl">
+      <div className="card-body">
+        <h1 className="card-title pt-1/15">My Saved Psychiatrists</h1>
+        <div className="grid grid-cols-4 gap-4 items-center pb-1/12 shrink">
+          <PyschiatristCard p_name={names[0]} p_certifications="Certifications Certifications" size={1}></PyschiatristCard>
+          <PyschiatristCard p_name={names[1]} p_certifications="Certifications Certifications" size={1}></PyschiatristCard>
+          <PyschiatristCard p_name={names[2]} p_certifications="Certifications Certifications" size={1}></PyschiatristCard>
+          <PyschiatristCard p_name={names[3]} p_certifications="Certifications Certifications" size={1}></PyschiatristCard>
```
Comment (sriramanved) [inline frontend/src/components/Dashboard.tsx]: same issue as above.

File: frontend/src/components/Dashboard.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    <div className="card w-full bg-base-100 shadow-xl">
+      <div className="card-body">
+        <h1 className="card-title pt-1/15">Additional Resources</h1>
+          <div className="grid grid-cols-4 gap-4 items-center pb-1/12 shrink"></div>
+            <div className="grid grid-cols-4 gap-4 margin-top: 100 items-center pb-1/12 shrink">
+              <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+              <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+              <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+              <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+            </div>
```
Comment (sriramanved) [inline frontend/src/components/Dashboard.tsx]: same issue as above.

File: frontend/src/components/Dashboard.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      </div>
+    </div>
+
+  </div >
+
+
+  // return <div> 
+  //    <AppointmentCard p_name = 'Erica Smith' start = {new Date('2016-01-04 13:30')} end = {new Date('2016-01-04 14:30')}>
+  //    </AppointmentCard>
+  // </div>;
```
Comment (sriramanved) [inline frontend/src/components/Dashboard.tsx]: remove uncommented code.

File: frontend/src/components/NavigationButtons.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
+
+const NavigationButtons = ({ goBack, goNext }) => {
```
Comment (sriramanved) [inline frontend/src/components/NavigationButtons.tsx]: why did you guys leave files from another issue in your PR? If you have dependency issues, its not a good idea to just simply clone someone else's working branch and then work off of that, leaving all of their work in your working branch when you make a PR. This code has nothing to do with the task at hand. Please remove.

File: frontend/src/components/QuestionCard.tsx [hunk 12]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (sriramanved) [inline frontend/src/components/QuestionCard.tsx]: This file should not be here.

File: frontend/src/components/QuestionnaireCarousel.tsx [hunk 13]
```diff
@@ -0,0 +1,81 @@
+import { useState } from "react";
```
Comment (sriramanved) [inline frontend/src/components/QuestionnaireCarousel.tsx]: This file should not be here.

File: frontend/src/components/PsychiatristCard.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+          {isShown && <div className="card w-96 bg-base-100 shadow-xl">
+            <div className="card-body">
+              <div className="card-actions justify-end">
+                <button className="btn btn-square btn-sm " onClick={handleClick}>
+                  <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="current"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M6 18L18 6M6 6l12 12" /></svg>
+                </button>
+              </div>
+              <h2 className="card-title">{p_name}</h2>
+              <p>Pyschiatrist Profile</p>
```
Comment (sriramanved) [inline frontend/src/components/PsychiatristCard.tsx]: You need to comment your code.

File: frontend/src/components/PsychiatristCard.tsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+  const handleClick = event => {
+    setIsShown(!isShown);
+  };
+
+
+  return (
+    <div className="card w-11/12 bg-base-100 shadow-xl">
+      <div className="card-body">
+        <div><PsychiatristIcon></PsychiatristIcon></div>
```
Comment (sriramanved) [inline frontend/src/components/PsychiatristCard.tsx]: Render only `<PsychiatristIcon />`

File: frontend/src/index.d.ts [hunk 16]
```diff
@@ -0,0 +1 @@
+declare module '*.svg';
```
Comment (sriramanved) [inline frontend/src/index.d.ts]: why do you need this?

File: frontend/src/pages/_app.tsx [hunk 17]
```diff
@@ -1,11 +1,37 @@
...
+
+    //       <Route path="home" index element={<Home />} />
+    //       {/* <Route path="appointment" element={<Appt />} />
+    //     <Route path="heroes/:id" element={<Appt />} /> */}
+
+    //     </Route>
+    //   </Routes>
+    // </BrowserRouter>
+
+    // </BrowserRouter >
```
Comment (sriramanved) [inline frontend/src/pages/_app.tsx]: remove

File: frontend/src/pages/_app.tsx [hunk 18]
```diff
@@ -1,11 +1,37 @@
 import '@/styles/globals.css'
 import type { AppProps } from 'next/app'
+import ReactDOM from "react-dom";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import Home from '../pages/index';
+import React from 'react';
+import Appt from './appointment';
+import ApptDetails from '../components/ApptDetails'
+
```
Comment (sriramanved) [inline frontend/src/pages/_app.tsx]: why is this here?

File: frontend/src/pages/appointment.tsx [hunk 19]
```diff
@@ -0,0 +1,7 @@
+
+
+const Appt = () => {
+  return <h1>Hello world</h1>
+};
+
+export default Appt;
```
Comment (sriramanved) [inline frontend/src/pages/appointment.tsx]: What is this for?

File: frontend/src/pages/index.tsx [hunk 20]
```diff
@@ -1,123 +1,28 @@
...
-import Image from 'next/image'
-import { Inter } from 'next/font/google'
-import styles from '@/styles/Home.module.css'
+import Head from 'next/head';
+import React from 'react';
+
+import QuestionCard from '@/components/QuestionCard';
+import QuestionnaireCarousel from '@/components/QuestionnaireCarousel';
+import NavigationButtons from '@/components/NavigationButtons';
+import Dashboard from '@/components/Dashboard';
```
Comment (sriramanved) [inline frontend/src/pages/index.tsx]: You need to remove stuff you that you guys aren't using

File: frontend/src/pages/index.tsx [hunk 21]
```diff
@@ -1,123 +1,28 @@
...
-  )
+      <Dashboard></Dashboard>
+      {/* <QuestionCard
+        question="How are you feeling about your mental health?"
+        option1='Feeling great!'
+        option2='Could be better.'
+        option3="I'm not really sure"
+        option4="I'm not doing too well"
+        option5="I need some help"
+      /> */}
```
Comment (sriramanved) [inline frontend/src/pages/index.tsx]: This file was supposed to be left alone when you made the PR.

**Review by sriramanved — CHANGES_REQUESTED** [review]
Quite a few comments that need addressing. This code is not PR-ready. In the future, if you need more time to either learn about the task at hand and relevant concepts, please message me earlier in the week or come to work session and we can work it out. This ticket is unfinished. Even if you have visually implemented the mid-fis, I certainly cannot approve these changes until you have written clean code.

File: frontend/next.config.js (hunk 1 above)
Comment (jessicajcho) [reply frontend/next.config.js]: same as Dylan/Jae. allows for svg imgs to be added

File: frontend/src/index.d.ts (hunk 16 above)
Comment (jessicajcho) [reply frontend/src/index.d.ts]: it's related to the webpack installing. it was for adding svg images

File: frontend/src/components/Dashboard.tsx [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+        <div className="grid grid-cols-4 gap-4 margin-top: 100 items-center pb-1/12 shrink">
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+        </div>
+      </div>
+    </div>
+
+  </div >
```
Comment (Archit404Error) [inline frontend/src/components/Dashboard.tsx]:
```suggestion
  </div>
```

File: frontend/src/components/Dashboard.tsx [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+
+    <div className="card w-full bg-base-100 shadow-xl">
+      <div className="card-body">
+        <h1 className="card-title pt-1/15">Additional Resources</h1>
+        <div className="grid grid-cols-4 gap-4 items-center pb-1/12 shrink"></div>
+        <div className="grid grid-cols-4 gap-4 margin-top: 100 items-center pb-1/12 shrink">
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
+          <ArticleCard title={dummy_title} body={dummy_descrip}></ArticleCard>
```
Comment (Archit404Error) [inline frontend/src/components/Dashboard.tsx]: We should ideally be loading the data from somewhere and using `.map()` to generate these components -- I know right now y'all don't have a backend but you can also load the data from a JSON file and use `map()`, which can dynamically generate components as opposed to hard coding

File: frontend/src/components/Dashboard.tsx [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+        <h1 className="card-title pt-1/15">Upcoming Appointments</h1>
+        <div className="grid grid-cols-4 gap-4 items-center pb-1/12 shrink">
+          <AppointmentCard p_name={names[0]} start={new Date(starts[0])} end={new Date(ends[0])} description={dummy_descrip}>
+          </AppointmentCard>
+          <AppointmentCard p_name={names[1]} start={new Date(starts[1])} end={new Date(ends[1])} description={dummy_descrip}>
+          </AppointmentCard>
+          <AppointmentCard p_name={names[2]} start={new Date(starts[2])} end={new Date(ends[2])} description={dummy_descrip}>
+          </AppointmentCard>
+          <AppointmentCard p_name={names[3]} start={new Date(starts[3])} end={new Date(ends[3])} description={dummy_descrip}>
+          </AppointmentCard>
```
Comment (Archit404Error) [inline frontend/src/components/Dashboard.tsx]: Again, use `map()` here and load data from a JSON file

File: frontend/src/components/Dashboard.tsx [hunk 25]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import AppointmentCard from './AppointmentCard';
+import PyschiatristCard from './PsychiatristCard';
+import ArticleCard from './ArticleCard';
+
+const names = ["Erica Smithson", "Ronald River", "Hermes Heroic", "Pomelo Pigskin"]
+const starts = ['2023-04-10 13:30', '2023-05-10 12:30', '2023-06-15 14:15', '2023-06-20 14:15']
+const ends = ['2023-04-10 14:30', '2023-05-10 13:30', '2023-06-15 15:15', '2023-06-20 15:15']
+const dummy_descrip = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam dignissim efficitur sapien id facilisis. In imperdiet ante ac fermentum aliquet. Nam pellentesque vestibulum nisl quis facilisis. Sed eget magna auctor."
+const dummy_title = "This is the title of a mental health resource article";
```
Comment (Archit404Error) [inline frontend/src/components/Dashboard.tsx]: All this data can be represented in a dummy JSON file rather than hard coding into your file directly. Then you can use `map()` to dynamically generate components. This will simulate how you transform API responses from the backend into components on the frontend.

File: frontend/src/components/AppointmentCard.tsx [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+
+          </div>
+        </div>
+      </div>}
+
+    </div>
```
Comment (Archit404Error) [inline frontend/src/components/AppointmentCard.tsx]:
```suggestion
    </div>
```

File: frontend/src/components/AppointmentCard.tsx [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+            <div className="col-span-4"><p>{start.getHours()}:{start.getMinutes()} - {end.getHours()}:{end.getMinutes()}</p></div>
+          </div>
+
+          <div className="card-actions justify-center">
+            <button className="btn w-7/12" >More Info</button>
+
+
+          </div>
+        </div>
+      </div>}
```
Comment (Archit404Error) [inline frontend/src/components/AppointmentCard.tsx]: Why is this in JSX curly braces?

You can just write it without curly braces as such:
```suggestion
      <div className="card w-11/12 bg-base-100 shadow-xl">
        <div className="card-body">
          <p>In {daysTo} days</p>
          <h2 className="card-title">Meeting with {p_name}</h2>

          <div className="grid grid-cols-5 grid-rows-2 gap-4 items-center pb-1/12">
            <div className="col-span-1 shrink"><CalendarIcon></CalendarIcon></div>
            <div className="col-span-4"><p>{day}, {month} {start.getDay()}</p></div>
            <div className="col-span-1 shrink"><ClockIcon></ClockIcon></div>
            <div className="col-span-4"><p>{start.getHours()}:{start.getMinutes()} - {end.getHours()}:{end.getMinutes()}</p></div>
          </div>

          <div className="card-actions justify-center">
            <button className="btn w-7/12" >More Info</button>


          </div>
        </div>
      </div>
```

File: frontend/src/pages/index.tsx [hunk 28]
```diff
@@ -1,123 +1,17 @@
...
-                alt="Vercel Logo"
-                className={styles.vercelLogo}
-                width={100}
-                height={24}
-                priority
-              />
-            </a>
-          </div>
-        </div>
+      <Dashboard></Dashboard>
```
Comment (Archit404Error) [inline frontend/src/pages/index.tsx]: Since there are no children for this component you can write it as such:
```suggestion
      <Dashboard />
```

File: yarn.lock
Comment (Archit404Error) [inline yarn.lock:1]: Why do you have a `package.json` and `yarn.lock` both inside the frontend and root folders? You should **only** have `package.json` and `yarn.lock` in `frontend/`, please remove the other files

**Review by Archit404Error — CHANGES_REQUESTED** [review]
Good work so far, but there's some necessary fixes you guys will have to do for the sake of code quality -- most importantly, make sure to remove `package.json` and `yarn.lock` from root directory -- we should only have these files in `frontend/` and `backend/`

File: frontend/src/components/AppointmentCard.tsx (hunk 27 above)
Comment (Archit404Error) [reply frontend/src/components/AppointmentCard.tsx]: Also you should remove the extra whitespace here

File: frontend/src/components/ArticleCard.tsx (hunk 5 above)
Comment (jessicajcho) [reply frontend/src/components/ArticleCard.tsx]: are you referring specifically to ArticleIcon?

**Review by sriramanved — APPROVED** [review]
Good job on this PR! Remember to pass in props in a clean, modular way in order to test your component in the future, as you did with the .map() function.
